packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -193,6 +193,47 @@ exports.createContentTypeNodes = ({
       // Get localized fields.
       const entryItemFields = _.mapValues(entryItem.fields, v => getField(v))
 
+      // If entry is not set by user, provide an empty value of the same type
+      const setBlankValue = contentTypeItemField => {
+        if (contentTypeItemField.type.match(/Symbol|Text|Date/)) {
+          return ``
+        } else if (contentTypeItemField.type.match(/Number/)) {
+          return NaN
+        } else if (
+          contentTypeItemField.type.match(
+            /Object|Location|Media|Reference|Link/
+          )
+        ) {
+          return {}
+        } else if (contentTypeItemField.type.match(/Array/)) {
+          // Setting values recursively is useful for 'never defined entries'
+          // TODO: Does not work for arrays of references, assets, objects, ...
+          return [setBlankValue(contentTypeItemField.items)]
+        } else if (contentTypeItemField.type.match(/Boolean/)) {
+          return false
+        }
+      }
+      contentTypeItem.fields.forEach(contentTypeItemField => {
+        const fieldName = contentTypeItemField.id
+        if (typeof entryItemFields[fieldName] === `undefined`) {
+          entryItemFields[fieldName] = setBlankValue(contentTypeItemField)
+        }
+        // Add a "json" property on Objects with the stringified version of its content
+        // This allows for querying objects with changing properties
+        // TODO: how to handle a property "json" entered by user? Should we just
+        // use a more specific name than "json" for our created property
+        if (contentTypeItemField.type === `Object`) {
+          if (typeof entryItemFields[fieldName].json === `undefined`) {
+            entryItemFields[fieldName].json = JSON.stringify(
+              entryItemFields[fieldName]
```

Comment:
this is a proposition to solve #1703

Reply:
instead of converting every object to string, I added a json property to every object containing the stringified version of the object. This means we can query that json field on any object even if the object content vary from entry to entry.
Does that make sense?
Should we choose a wiser property name?